### PR TITLE
 feat(CompoundExpansion): add compound expansion example to table

### LINKED ISSUE #721: Data Table Enhancement: Compound Expansion
Enhance the Data Table to support compound expansion. 

List + Specs: https://drive.google.com/drive/u/1/folders/1cKk-HnEJFcB4w6sgl05NeLPVLRUUiNUW

Related issue #198

Component lifecycle: 
- [x] Interaction design requirements 
- [x] Visuals 
- [x] First checkpoint complete and approved 
- [ ] CSS 

[Compound Expansion - Expansion Hover.pdf](https://github.com/patternfly/patternfly-next/files/2650769/Compound.Expansion.-.Expansion.Hover.pdf)
[compound-expansion-mobile.pdf](https://github.com/patternfly/patternfly-next/files/2650770/compound-expansion-mobile.pdf)

### PR DESCRIPTION
**Pull request by christiemolloy**
Closes #721 and #1046 

- Creates the compound expansion table example. 
- Refactors the original table component by splitting code into desktop/mobile.
- Refines visuals for column headers when sortable columns are supported

### CODE REVIEW

**Comment by christiemolloy** [comment]
After speaking with @maryshak1996 she is going to revise some of the mobile designs also: 
- trs should have a margin-top and the border-bottom should be removed (christie).

File: src/patternfly/components/Table/table-th.hbs [hunk 1]
```diff
@@ -12,7 +17,7 @@
     {{{table-th--attribute}}}
   {{/if}}>
   {{#if table-th--sortable}}
-    <button>
+    {{#> button}}
```
Comment (mcoker) [inline src/patternfly/components/Table/table-th.hbs]: Do you mean to use the button component here? If so you need to use it with a modifier.

Comment (christiemolloy) [reply src/patternfly/components/Table/table-th.hbs]: updated, thanks!

**Comment by christiemolloy** [comment]
changes:
kebab now inline with text
added scroll onto to the compound expansion on mobile
got rid of weird background behind the first th
remove divider under first row in compound expansion
ungrouped css vars

**Pull request merged**